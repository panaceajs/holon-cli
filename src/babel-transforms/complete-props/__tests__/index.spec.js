const pluginTester = require('babel-plugin-tester');
const completeProps = require('../');
const pluginSyntaxObjectRestSpread = require('@babel/plugin-syntax-object-rest-spread');
const pluginJsx = require('@babel/plugin-syntax-jsx');
const dynamicImport = require('@babel/plugin-syntax-dynamic-import');

pluginTester({
  filename: __filename,
  plugin: completeProps,
  snapshot: true,
  babelOptions: {
    sourceType: 'module',
    plugins: [dynamicImport, pluginJsx, pluginSyntaxObjectRestSpread]
  },
  tests: [
    {
      title: 'it should add nothing when there is nothing to add.',
      fixture: '__fixtures__/sfc-without-existing-props.js',
      pluginOptions: {}
    },
    {
      title: 'it should add props.',
      fixture: '__fixtures__/sfc-without-existing-props.js',
      pluginOptions: { props: ['first', 'second'] }
    },
    {
      title: 'it should add prop functions.',
      fixture: '__fixtures__/sfc-without-existing-props.js',
      pluginOptions: { dispatchProps: ['doSomething', 'doSomethingElse'] }
    },
    {
      title: 'it should not overwrite existing props.',
      fixture: '__fixtures__/sfc-with-existing-props.js',
      pluginOptions: { props: ['existing', 'nonExisting'] }
    },
    {
      title: 'it should not overwrite existing prop functions.',
      fixture: '__fixtures__/sfc-with-existing-props.js',
      pluginOptions: { dispatchProps: ['existingDoSomething'] }
    },
    {
      title: 'it should set both props and prop functions.',
      fixture: '__fixtures__/sfc-with-existing-props.js',
      pluginOptions: {
        props: ['existing', 'non-existing'],
        dispatchProps: ['existingDoSomething', 'nonExistingDoSomething']
      }
    },
    {
      title:
        'it should rename given props to variable names and remove duplicates.',
      fixture: '__fixtures__/sfc-without-existing-props.js',
      pluginOptions: {
        props: [
          'this is a var',
          'this_is_a_var',
          'thisIsAVar',
          'this_is_another_var',
          'this.also'
        ],
        dispatchProps: [
          'existingDoSomething',
          'nonExistingDoSomething',
          'this is a var',
          'this_is_a_var_',
          'this_is_a_var_1'
        ]
      }
    }
  ]
});
