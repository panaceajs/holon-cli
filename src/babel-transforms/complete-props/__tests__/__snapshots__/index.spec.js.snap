// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`holon-complete-props it should add nothing when there is nothing to add.: it should add nothing when there is nothing to add. 1`] = `
"
import React from 'react';

export default () => (<div>123</div>);

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

export default (() => <div>123</div>);
"
`;

exports[`holon-complete-props it should add prop functions.: it should add prop functions. 1`] = `
"
import React from 'react';

export default () => (<div>123</div>);

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

export default (({
  doSomething = () => {},

  doSomethingElse = () => {}

}) => <div>123</div>);
"
`;

exports[`holon-complete-props it should add props.: it should add props. 1`] = `
"
import React from 'react';

export default () => (<div>123</div>);

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

export default (({
  first,
  second
}) => <div>123</div>);
"
`;

exports[`holon-complete-props it should not overwrite existing prop functions.: it should not overwrite existing prop functions. 1`] = `
"
import React from 'react';

export default ({ existing = 10, existingDoSomething = () => { /* yes we exist */ }}) => (<div>123</div>);

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

export default (({ existing = 10, existingDoSomething = () => {/* yes we exist */} }) => <div>123</div>);
"
`;

exports[`holon-complete-props it should not overwrite existing props.: it should not overwrite existing props. 1`] = `
"
import React from 'react';

export default ({ existing = 10, existingDoSomething = () => { /* yes we exist */ }}) => (<div>123</div>);

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

export default (({ existing = 10, existingDoSomething = () => {/* yes we exist */}, nonExisting
}) => <div>123</div>);
"
`;

exports[`holon-complete-props it should rename given props to variable names and remove duplicates.: it should rename given props to variable names and remove duplicates. 1`] = `
"
import React from 'react';

export default () => (<div>123</div>);

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

export default (({
  thisIsAVar,
  thisIsAnotherVar,
  thisAlso,

  existingDoSomething = () => {},

  nonExistingDoSomething = () => {},

  thisIsAVar1 = () => {}

}) => <div>123</div>);
"
`;

exports[`holon-complete-props it should set both props and prop functions.: it should set both props and prop functions. 1`] = `
"
import React from 'react';

export default ({ existing = 10, existingDoSomething = () => { /* yes we exist */ }}) => (<div>123</div>);

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

export default (({ existing = 10, existingDoSomething = () => {/* yes we exist */}, nonExisting,

  nonExistingDoSomething = () => {}

}) => <div>123</div>);
"
`;
